### PR TITLE
Add 2 min timeout to DB properties

### PR DESCRIPTION
**Pull request by Anviking**
# Issue Number

#2292 and related failures.

# Overview

- [x] Add 2 min timeout to DB properties

# Comments

To help debug timeout failures. If they fail properly, we might get useful seeds etc.

### CODE REVIEW

File: lib/core/test/unit/Cardano/Wallet/DB/Properties.hs [hunk 1]
```diff
@@ -492,12 +493,12 @@ prop_removeWalletTwice
...
     -> Property
 prop_removeWalletTwice db@DBLayer{..} (key@(PrimaryKey wid), cp, meta) =
-    monadicIO (setup >> prop)
+    within (2 * minute) $ monadicIO (setup >> prop)
   where
     setup = liftIO $ do
         cleanDB db
         atomically $ unsafeRunExceptT $ initializeWallet key cp meta mempty pp
-    prop = liftIO $ do
+    prop =  liftIO $ do
```
Comment (jonathanknowles) [inline lib/core/test/unit/Cardano/Wallet/DB/Properties.hs]: A bit of stray space found its way into the line:

```suggestion
    prop = liftIO $ do
```

File: lib/core/test/unit/Cardano/Wallet/DB/Properties.hs [hunk 2]
```diff
@@ -729,7 +730,7 @@ prop_sequentialPut
         -- ^ Property arguments
     -> Property
 prop_sequentialPut putOp readOp resolve db@DBLayer{..} kv =
-    cover 25 cond "conflicting db entries" $ monadicIO (setup >> prop)
+    cover 25 cond "conflicting db entries" $ within (2 * minute) $ monadicIO (setup >> prop)
```
Comment (jonathanknowles) [inline lib/core/test/unit/Cardano/Wallet/DB/Properties.hs]: Request: wrap this, as it's quite a bit over the line length limit. :cat: 

```suggestion
    cover 25 cond "conflicting db entries"
        $ within (2 * minute)
        $ monadicIO (setup >> prop)
```

File: lib/core/test/unit/Cardano/Wallet/DB/Properties.hs [hunk 3]
```diff
@@ -769,7 +770,7 @@ prop_parallelPut
         -- ^ Property arguments
     -> Property
 prop_parallelPut putOp readOp resolve db@DBLayer{..} (KeyValPairs pairs) =
-    cover 25 cond "conflicting db entries" $ monadicIO (setup >> prop)
+    cover 25 cond "conflicting db entries" $ within (2 * minute) $ monadicIO (setup >> prop)
```
Comment (jonathanknowles) [inline lib/core/test/unit/Cardano/Wallet/DB/Properties.hs]: Request: wrap this, as it's quite a bit over the line length limit. :cat: 

```suggestion
    cover 25 cond "conflicting db entries"
        $ within (2 * minute)
        $ monadicIO (setup >> prop)
```

File: lib/core/test/unit/Cardano/Wallet/DB/Properties.hs [hunk 4]
```diff
@@ -1068,3 +1069,7 @@ genBatches (GenSparseCheckpointsArgs cfg h) = do
         -- sure we generate realistic cases.
         n <- fromIntegral <$> choose (1, 3 * gapSize cfg)
         go (drop n source) (take n source : batches)
+
+-- | Microseconds in a minute
+minute :: Int
+minute = 60*1000*1000
```
Comment (jonathanknowles) [inline lib/core/test/unit/Cardano/Wallet/DB/Properties.hs]: Suggestion: use `NumericUnderscores` (and use a plural "minutes").

```suggestion
minutes :: Int
minutes = 60_000_000
```

**Review by jonathanknowles — APPROVED** [review]
Hi @Anviking 

I have a few suggestions and a couple of requests. (See comments.)

But otherwise looks good!

File: lib/core/test/unit/Cardano/Wallet/DB/Properties.hs (hunk 4 above)
Comment (Anviking) [reply lib/core/test/unit/Cardano/Wallet/DB/Properties.hs]: NumericUnderscores — sure. Not sure we should use plural though? 

I know we have decided to use singular for units in the context of `Quantity` at least:

```shell
$ git grep "Quantity \"slot"                                                                                                                                                                                                                               
lib/core/src/Cardano/Wallet/Api/Types.hs:    , epochLength :: !(Quantity "slot" Word32)
lib/core/src/Cardano/Wallet/Primitive/Slotting.hs:slotDifference :: SlotParameters -> SlotId -> SlotId -> Quantity "slot" Natural
lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs:instance Arbitrary (Quantity "slot" Word32) where
lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs:instance Arbitrary (Quantity "slot" Natural) where
```

**Comment by Anviking** [comment]
Hmmmm. That `within` didn't seem to help 😄 Interesting.

**Comment by Anviking** [comment]
Closing since it didn't help

**Pull request closed without merging**